src/entities/addLiquidity/index.ts

```diff
@@ -33,14 +42,26 @@ export class AddLiquidity implements AddLiquidityBase {
     }
 
     buildCall(input: AddLiquidityBuildCallInput): AddLiquidityBuildCallOutput {
-        if (input.vaultVersion === 2 && 'sender' in input) {
-            const addLiquidity = new AddLiquidityV2(this.config);
-            return addLiquidity.buildCall(input);
-        }
-
-        if (input.vaultVersion === 3 && !('sender' in input)) {
-            const addLiquidity = new AddLiquidityV3();
-            return addLiquidity.buildCall(input);
+        switch (input.vaultVersion) {
+            case 0: {
+                // TODO: check if we need to be more restrictive with input type mismatch here
```

Comment:
I guess we don't do any validation for V2/V3 because we assume buildCall input comes from `query` where validation has already been done so probably ok to keep this assumption.

Reply:
done ✅